storage: GenericEphemeralVolume (#22438)
kubernetes/website: content/en/docs/concepts/storage/volumes.md

> may use the `csi` volume type to attach, mount, etc. the volumes exposed by the

Reviewer: Should we link to the new ephemeral-volumes page somewhere at the top of this page? (and link to the persistent volumes page in the same area)

Author: The "background" section may need a more thorough rewrite. For example, it still says:
> A Kubernetes volume, on the other hand, has an explicit lifetime - the same as
the Pod that encloses it.

That became obsolete when persistent volumes were introduced, which was a while ago...

Let me punt on that for now for the sake of getting this PR merged before the docs deadline. I've created https://github.com/kubernetes/website/issues/22505 to track it.

CSI driver.

A `csi` volume can be used in a pod in three different ways:
- through a reference to a [`persistentVolumeClaim`](#persistentvolumeclaim)

> - with a [generic ephemeral volume](/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volume)

Reviewer: should we note that these are alpha/beta respsectively?

Author: Okay.

{{< feature-state for_k8s_version="v1.16" state="beta" >}}

> This feature allows CSI volumes to be directly embedded in the Pod specification instead of a PersistentVolume. Volumes specified in this way are ephemeral and do not persist across Pod restarts. See [the ephemeral volume page](/docs/concepts/storage/ephemeral-volumes/#csi-ephemeral-volume) for more information.

Reviewer 2: ```suggestion
You can directly configure CSI volume claims within the Pod specification. Volumes specified in this way are ephemeral and do not persist across Pod restarts. See [Ephemeral Volumes](/docs/concepts/storage/ephemeral-volumes/#csi-ephemeral-volume) for more information.
```

I'm saying _volume claim_ because “ephemeral PersistentVolumeClaim” doesn't really make sense.

Author: In this case "CSI volumes" makes more sense than "CSI volume claims" because there is nothing resembling a "claim" in the API. The rest makes sense, changed.

#### Developer resources
For more information on how to develop a CSI driver, refer to the [kubernetes-csi

> documentation](https://kubernetes-csi.github.io/docs/)

Reviewer 2: ```suggestion
 ## {{% heading "whatsnext" %}}

For more information on how to develop a CSI driver, refer to the [kubernetes-csi documentation](https://kubernetes-csi.github.io/docs/).
```
